test/e2e/common/setup.go

```diff
@@ -0,0 +1,310 @@
+	"k8s.io/utils/env"
+)
+
+var (
+	clusterNames = []string{"east", "west"}
+
+	AppNs    namespace.Instance
+	EastApps echo.Instances
+	WestApps echo.Instances
+
+	_, file, _, _ = runtime.Caller(0)
+	rootDir       = filepath.Join(filepath.Dir(file), "../../..")
+
+	testHub = env.GetString("HUB", "quay.io/maistra-dev")
+	testTag = env.GetString("TAG", "latest")
+
+	istioVersion = env.GetString("ISTIO_VERSION", "1.22.1")
+)
+
+const (
+	EastClusterName = "cluster-0"
+	WestClusterName = "cluster-1"
+)
+
+func CreateControlPlaneNamespace(ctx resource.Context) error {
+	if len(ctx.Clusters()) > 2 {
+		return fmt.Errorf("too many clusters - expected 2, got %d", len(ctx.Clusters()))
+	}
+
+	createNamespace := func(cluster cluster.Cluster) error {
+		if _, err := cluster.Kube().CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{
+			ObjectMeta: v1.ObjectMeta{
+				Name: "istio-system",
+			},
+		}, v1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create namespace: %w", err)
+		}
+		return nil
+	}
+
+	for _, c := range ctx.Clusters() {
+		if err := retry.UntilSuccess(func() error {
+			_, err := c.Kube().CoreV1().Namespaces().Get(context.Background(), "istio-system", v1.GetOptions{})
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return fmt.Errorf("failed to get namespace: %w", err)
+				}
+				return createNamespace(c)
+			}
+			if err := c.Kube().CoreV1().Namespaces().Delete(context.Background(), "istio-system", v1.DeleteOptions{}); err != nil {
+				return fmt.Errorf("failed to delete namespace: %w", err)
+			}
+			return createNamespace(c)
+		}, retry.Timeout(30*time.Second), retry.Delay(1*time.Second)); err != nil {
+			return err
+		}
+	}
+
+	ctx.Cleanup(func() {
+		for idx, c := range ctx.Clusters() {
+			if err := c.Kube().CoreV1().Namespaces().Delete(context.Background(), "istio-system", v1.DeleteOptions{}); err != nil {
+				scopes.Framework.Errorf("failed to delete control plane namespace (cluster=%s): %w", clusterNames[idx], err)
+			}
+		}
+	})
+	return nil
+}
+
+func CreateCACertsSecret(ctx resource.Context) error {
+	for idx, c := range ctx.Clusters() {
+		clusterName := clusterNames[idx]
+		data := map[string][]byte{
+			"root-cert.pem":  {},
+			"cert-chain.pem": {},
+			"ca-cert.pem":    {},
+			"ca-key.pem":     {},
+		}
+		if err := setCacertKeys(fmt.Sprintf("%s/test/testdata/certs/%s", rootDir, clusterName), data); err != nil {
+			return fmt.Errorf("failed to set keys in cacerts secret (cluster=%s): %w", clusterName, err)
+		}
+		cacerts := &corev1.Secret{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "cacerts",
+				Namespace: "istio-system",
+			},
+			Data: data,
+		}
+		if _, err := c.Kube().CoreV1().Secrets("istio-system").Create(context.Background(), cacerts, v1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create cacerts secret (cluster=%s): %w", clusterName, err)
+		}
+	}
+	return nil
+}
+
+func setCacertKeys(dir string, data map[string][]byte) error {
```

Comment:
bundle with the one above

```suggestion
func setCACertKeys(dir string, data map[string][]byte) error {
```